feature/staff/src/main/kotlin/io/github/droidkaigi/confsched2023/staff/StaffListItem.kt

```diff
@@ -0,0 +1,105 @@
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Person
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.vector.rememberVectorPainter
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import io.github.droidkaigi.confsched2023.model.Staff
+import io.github.droidkaigi.confsched2023.ui.previewOverride
+import io.github.droidkaigi.confsched2023.ui.rememberAsyncImagePainter
+
+private val staffIconShape = RoundedCornerShape(20.dp)
+
+@Composable
+fun StaffListItem(
+    staff: Staff,
+    onStaffClick: (url: String) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier
+            .clickable(enabled = staff.profileUrl.isNotEmpty()) {
+                onStaffClick(staff.profileUrl)
+            }
+            .padding(horizontal = 16.dp, vertical = 10.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Image(
+            painter = previewOverride(previewPainter = { rememberVectorPainter(image = Icons.Default.Person) }) {
+                rememberAsyncImagePainter(staff.iconUrl)
+            },
+            contentDescription = null,
+            modifier = Modifier
+                .size(60.dp)
+                .clip(staffIconShape)
+                .border(
+                    width = 1.dp,
+                    color = MaterialTheme.colorScheme.outline,
+                    shape = staffIconShape
+                ),
+        )
+        Spacer(modifier = Modifier.size(23.dp))
+        Text(
+            text = staff.username,
+            style = MaterialTheme.typography.bodyLarge,
+            maxLines = 2,
+            overflow = TextOverflow.Ellipsis,
+        )
+    }
+}
+
+// region Preview
+
+class PreviewStaffProvider : PreviewParameterProvider<Staff> {
+    override val values: Sequence<Staff>
+        get() = sequenceOf(
+            Staff(
+                id = 1,
+                username = "",
+                iconUrl = "",
+                profileUrl = ""
+            ),
+            Staff(
+                id = 1,
+                username = "UserName",
+                iconUrl = "",
+                profileUrl = ""
+            ),
+            Staff(
+                id = 1,
+                username = "UserNameUserNameUserNameUserNameUserNameUserNameUserName",
+                iconUrl = "",
+                profileUrl = ""
+            ),
+        )
+}
+
+@Preview(showBackground = true)
```

Comment:
Question.
It seems that Preview's public Composable function is being WARNED by the compose-lint-rule.
Looking at other modules, it appears that Preview's composable is not warning even though it is public.
Is there something wrong with the module settings? ?

```
Composables annotated with @Preview that are used only for previewing the UI should not be public.
```

Reply:
If set to private, showkase seems to throw a build error.

```
The methods annotated with Preview can't be private as Showkase won't be able to access them otherwise. If you'd like to skip this check and ignore the private previews, kindly pass skipPrivatePreviews=true as an annotation processor option.To learn more about how to set this option, read the Showkase README here- https://github.com/airbnb/Showkase/blob/master/README.md
	at com.airbnb.android.showkase.processor.logging.ShowkaseValidator.validateComponentElementOrSkip$showkase_processor(ShowkaseValidator.kt:57)
	at com.airbnb.android.showkase.processor.ShowkaseProcessor.processPreviewAnnotation(ShowkaseProcessor.kt:115)
	at com.airbnb.android.showkase.processor.ShowkaseProcessor.processComponentAnnotation(ShowkaseProcessor.kt:83)
	at com.airbnb.android.showkase.processor.ShowkaseProcessor.process(ShowkaseProcessor.kt:66)
	at com.airbnb.android.showkase.processor.BaseProcessor.internalProcess(BaseProcessor.kt:96)
	at com.airbnb.android.showkase.processor.BaseProcessor.process(BaseProcessor.kt:81)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$8$1.invoke(KotlinSymbolProcessingExtension.kt:305)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$8$1.invoke(KotlinSymbolProcessingExtension.kt:303)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.handleException(KotlinSymbolProcessingExtension.kt:409)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:303)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:112)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:88)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:256)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:247)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:87)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:47)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:168)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:53)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:100)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:46)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1497)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(Unknown Source)
	at java.rmi/sun.rmi.transport.Transport$1.run(Unknown Source)
	at java.rmi/sun.rmi.transport.Transport$1.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

```

Reply:
We have to investigate it but we might use internal modifier here

Reply:
OK
Once I added the internal modifier, lint stopped throwing errors!